src/findings/findingsTreeDataProvider.ts

```diff
@@ -0,0 +1,476 @@
+        } else if (element.category === 'older') {
+          return new OlderIssuesNode();
+        }
+        return null;
+      }
+      return null;
+    }
+
+    // For FindingNode, find the parent file
+    const parentFile = this.getRootFiles().find(file => file.fileUri === element.fileUri);
+    if (parentFile) {
+      if (isFocusingOnNewCode()) {
+        return element.isOnNewCode ? new NewIssuesNode() : new OlderIssuesNode();
+      }
+      return parentFile;
+    }
+
+    return null;
+  }
+
+  private getRootItems(): FindingsTreeViewItem[] {
+    if (isFocusingOnNewCode()) {
+      const newIssuesFiles = this.getNewIssuesFiles();
+      const olderIssuesFiles = this.getOlderIssuesFiles();
+      
+      const items: FindingsTreeViewItem[] = [];
+      
+      if (newIssuesFiles.length > 0) {
+        items.push(new NewIssuesNode());
+      }
+      
+      if (olderIssuesFiles.length > 0) {
+        items.push(new OlderIssuesNode());
+      }
+      
+      return items;
+    } else {
+      return this.getRootFiles();
+    }
+  }
+
+  private getNewIssuesFiles(): FindingsFileNode[] {
+    const files: FindingsFileNode[] = [];
+    
+    this.findingsCache.forEach((findings, fileUri) => {
+      const newFindings = findings.filter(finding => finding.isOnNewCode);
+      if (newFindings.length > 0) {
+        files.push(new FindingsFileNode(fileUri, newFindings.length, 'new'));
+      }
+    });
+    
+    return files;
+  }
+
+  private getOlderIssuesFiles(): FindingsFileNode[] {
+    const files: FindingsFileNode[] = [];
+    
+    this.findingsCache.forEach((findings, fileUri) => {
+      const olderFindings = findings.filter(finding => !finding.isOnNewCode);
+      if (olderFindings.length > 0) {
+        files.push(new FindingsFileNode(fileUri, olderFindings.length, 'older'));
+      }
+    });
+    
+    return files;
+  }
+
+  getRootFiles(): FindingsFileNode[] {
+    const files: FindingsFileNode[] = [];
+    
+    this.findingsCache.forEach((findings, fileUri) => {
+      if (findings.length > 0) {
+        files.push(new FindingsFileNode(fileUri, findings.length));
+      }
+    });
+    
+    return files;
+  }
+
+  private getFindingsForFile(fileUri: string, category?: 'new' | 'older'): FindingNode[] {
+    const findings = this.findingsCache.get(fileUri) || [];
+    if (category) {
+      const lookingForNew = category === 'new';
+      // looking for new and is new, or looking for older and is older
+      return findings.filter(finding => finding.isOnNewCode === lookingForNew);
+    }
+    return findings.map(finding => new FindingNode(
+      finding.key,
+      finding.serverIssueKey,
+      finding.range,
+      finding.contextValue,
+      finding.source,
+      finding.message,
+      finding.ruleKey,
+      fileUri,
+      finding.status,
+      finding.findingType,
+      finding.isAiCodeFixable,
+      finding.isOnNewCode,
+      finding.vulnerabilityProbability,
+      finding.severity,
+    ));
```

Comment:
Could we imagine to pass the full `Finding` to the `FindingNode` instead of passing individual parameters?